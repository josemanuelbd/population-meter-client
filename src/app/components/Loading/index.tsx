import React from 'react';
import { Container } from './styles';

interface Props {
  text?: string;
}

const Loading: React.FC<Props> = ({ text = 'Loading...' }: Props) => <Container>{ text }</Container>;

export default Loading;
