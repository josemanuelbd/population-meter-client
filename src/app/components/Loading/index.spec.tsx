import React from 'react';
import { shallow } from 'enzyme';

import Loading from '.';

describe('Loading', () => {
  it('should return a custom loading', () => {
    const component = shallow(<Loading text="Custom loading" />);

    expect(component).toMatchSnapshot();
  });

  it('should return default loading', () => {
    const component = shallow(<Loading />);

    expect(component).toMatchSnapshot();
  });
});
