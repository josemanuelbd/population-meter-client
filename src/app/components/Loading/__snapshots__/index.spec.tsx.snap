// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should return a custom loading 1`] = `
<styles__Container>
  Custom loading
</styles__Container>
`;

exports[`Loading should return default loading 1`] = `
<styles__Container>
  Loading...
</styles__Container>
`;
